### PR TITLE
Add --save and --save-dev Options to clib-install

### LINKED ISSUE #19: Implement --save option for clib-install

### LINKED ISSUE #20: Implement --save-dev for clib-install

### PR DESCRIPTION
**Pull request by darthtrevino**
Hey Team;
I've added the --save and --save-dev options to clib-install. I needed to update the dependencies with the latest version of parson to support pretty JSON writing. The gist of the update is in `src/clib-install.c`. My C is a little rusty, so feel free to throw any feedback my way.

Thanks!
Chris

Fixes #19
Fixes #20

### CODE REVIEW

**Comment by darthtrevino** [comment]
Bah, my test is jacked. I'll post an update soon.

**Comment by darthtrevino** [comment]
Because of how parson persists JSON - forward-slashes in repos come out escaped, which is a little ugly. `clib install` will still work though.
 
```json
{
    "name": "test-package",
    "dependencies": {
        "stephenmathieson\/tabs-to-spaces": "1.0.0"
    }
}
```

File: package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "strdup": "0.0.0",
     "Constellation/console-colors.c": "1.0.1",
     "littlstar/asprintf.c": "0.0.3",
-    "logger": "0.0.0"
+    "logger": "0.0.0",
+    "kgabis/parson": "master"
```
Comment (stephenmathieson) [inline package.json]: i'd much rather see an actual version pinned here, or at least a commit hash

File: src/clib-install.c [hunk 2]
```diff
@@ -172,6 +186,52 @@ executable(clib_package_t *pkg) {
 
 #undef E_FORMAT
 
+char* concat(char *s1, char *s2) {
```
Comment (stephenmathieson) [inline src/clib-install.c]: this is a bit awkward.  can we either offload this to a dependency?

**Comment by stephenmathieson** [comment]
first, sorry for the _huge_ delay!

overall, this lgtm.  left a few comments inline though =)

File: package.json (hunk 1 above)
Comment (stephenmathieson) [inline package.json]: @darthtrevino i've forked the original `parson` repo and reverted the `\/` nonsense.  can you update to use [`clibs/parson@1.0.0`](https://github.com/clibs/parson/tree/1.0.0)?

Comment (darthtrevino) [inline package.json]: Sure! I apologize for not getting back to this PR, life got busy after the
holiday break. I'll make some time this week to polish it up.

On Mon, Jan 11, 2016 at 9:33 AM, Stephen Mathieson <notifications@github.com

Comment (stephenmathieson) [inline package.json]: no worries!  it took me forever to review your initial commits haha

**Comment by darthtrevino** [comment]
Hey Steve, 
I've addressed your comments. One thought I have is about the --save and --save-dev shorthand. Presently -s is shorthand for --save and -e is shorthand for --save-dev. If we want this to act kind of like node, it may make more sense to use -S and -D respectively.

**Comment by jwerle** [comment]
@darthtrevino Using `-S` and `-D` for shorthand makes sense to me!

**Comment by darthtrevino** [comment]
Thanks @jwerle! I was extending the tests to check the shorthands and ran into a bug. I'll try to have a push up soon to fix it.

`clib install -d jwerle/fs.c`

```
"development": {
    "jwerle/fs": {
        "c": "0.1.1"
    }
}
```

**Comment by jwerle** [comment]
@darthtrevino cool! yeah https://github.com/clibs/clib/commit/3397092e2be7a95b76f1a2ed06dda228e7939c36 looks good too

**Comment by stephenmathieson** [comment]
awesome!  thanks!!

**Comment by jb55** [comment]
yay! :confetti_ball:

**Pull request merged**